frontend/src/hooks/useUser.ts

```diff
@@ -0,0 +1,29 @@
+import { useQuery } from 'react-query'
+
+import { UserDto } from '~shared/types/user'
+
+import { LOGGED_IN_KEY } from '~constants/localStorage'
+import { fetchUser } from '~services/UserService'
+
+import { useLocalStorage } from './useLocalStorage'
+
+type UseUserReturn = {
+  user: UserDto | undefined
+  isLoading: boolean
+}
+
+export const useUser = (): UseUserReturn => {
+  const [isLoggedIn] = useLocalStorage<boolean>(LOGGED_IN_KEY)
+
+  const { data: user, isLoading } = useQuery<UserDto>(
+    'currentUser',
```

Comment:
how are we going to manage query keys when we have many many custom hooks? any reliable way to prevent duplication?

Reply:
Seems like a great article about this, wdyt: https://tkdodo.eu/blog/effective-react-query-keys

Reply:
Implemented in 12ed93f